### PR TITLE
Dplyr deprecated and CMHC data vignettes

### PR DESCRIPTION
**Pull request by mountainMath**
Small change to address dplyr deprecations. Two vignettes showing how to use CMHC data. This addresses https://github.com/mountainMath/cancensus/issues/140.

### CODE REVIEW

**Comment by mountainMath** [comment]
Added a minor change to also display the geo_dataset in `list_census_datasets`.

**Comment by dshkol** [comment]
I reworked the new vignettes a bit to flesh them out a bit. I think they could use a little bit more generalization. Some of the code there is very concise and cool on your part but would be very difficult to parse for a new reader and not really in the spirit of what a vignette should accomplish. Can we expand a little bit there so that a casual user can follow along? 

Also, do we need to add any license etc info?

**Comment by dshkol** [comment]
While we're at it, set up a new bootstrap theme that looks and fits much better with the overall feel of the site.

**Comment by mountainMath** [comment]
The licence information is a good question. I am thinking we should bake licence information into the package. Actually, right into CensusMapper. Maybe add a `licence` or `attribution` column in the `list_census_datasets`?

**Comment by dshkol** [comment]
Yeah, I think that approach works-  adding a licence or attribution col to `list_census_datasets()`. An alternative is to add it as an attribute. `attr(dataset,'licence')`.

**Comment by dshkol** [comment]
New version looks pretty good now. A few minor points on the taxfiler vignette:

1. This section is still a bit hard to parse out what is happening. I get it now after working through it, but might be a little bit over the top for a vignette. 
```
plot_data <- all_vectors %>%
  as.character() %>%
  gsub("^v_","",.)%>%
  gsub("_\\d+$","",.) %>%
  unique %>%
  sort %>%
  lapply(function(dataset) {
    vectors <- all_vectors[grepl(dataset,all_vectors)]
    d <- get_census(dataset,regions=list(CMA="59933"),vectors = vectors,
                    geo_format = 'sf', level="CT", quiet = TRUE) %>%
      select(c("GeoUID",names(vectors))) %>%
      pivot_longer(cols = matches("\\d{4}"),names_to = c("Year",".value"),names_pattern = "(\\d{4})_(.+)")
  }) %>%
  bind_rows() %>%
  sf::st_sf() %>%
  mutate(share=`CFLIM-AT`/Families)
```
2. The visualization doesn't render very well on vignettes because of the limited space for graphics. With that many facets, the underlying images all look compressed and more or less identical. 

3. Related to (1) and (2), I recommend either adding a more introductory basic example before going into this one, or just replacing the advanced one with a simple one. 

Once we merge this one and the other one in together, will need to go through the material again and make sure everything is aligned with the latest changes.

**Comment by mountainMath** [comment]
I have simplified the vignette, and I *think* it should be quite accessible now.

**Comment by dshkol** [comment]
Getting an error in the taxfiler vignette from the chunk starting on line 85

```
ggplot(plot_data,aes(fill=share)) +
  geom_sf(size=0.1,color="white") +
  facet_wrap("Year") +
  scale_fill_viridis_c(labels=scales::percent,option = "inferno",
                       trans="log",breaks = c(0.05,0.1,0.2,0.4)) +
  coord_sf(datum=NA,xlim=c(-123.4, -122.5), ylim=c(49.01, 49.4)) +
  labs(title="Share of census families in low income",fill="Share",
       caption=attribution)
```

```
Error in as.name(geometry_col) : invalid type/length (symbol/0) in vector allocation
```

**Comment by dshkol** [comment]
Thanks for adding the `attribution_for_datasets` function. Can I suggest renaming it to`dataset_attribution`? A bit shorter, and not using the passive voice.

**Comment by dshkol** [comment]
See new branch `dev` which has both new feature branches rolled into one
